qemu/tests/hotplug_virtio_mem.py

```diff
@@ -0,0 +1,57 @@
+import time
+
+from virttest import error_context
+
+from virttest.utils_misc import normalize_data_size
+from virttest.utils_test.qemu import MemoryHotplugTest
+from provider import virtio_mem_utils
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Virtio-mem hotplug test
+    1) Boot guest
+    2) Hotplug virtio-mem device
+    3) Check virtio-mem device
+    4) Resize virtio-mem device twice
+    5) Check virtio-mem device
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment
+    """
+
+    timeout = params.get_numeric("login_timeout", 240)
+    threshold = params.get_numeric("threshold", target_type=float)
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+    session = vm.wait_for_login(timeout=timeout)
+    virtio_mem_model = params.get("virtio_mem_model", "virtio-mem-pci")
+
+    target_mem = params.get("target_mems")
+    hotplug_test = MemoryHotplugTest(test, params, env)
+    hotplug_test.hotplug_memory(vm, target_mem)
+
+    for i, vmem_dev in enumerate(vm.devices.get_by_params({'driver': virtio_mem_model})):
+        device_id = vmem_dev.get_qid()
+        requested_size_vmem_test = params.get("requested-size_test_vmem%d" % i)
+        node_id = int(vmem_dev.params.get("node"))
+        virtio_mem_utils.check_memory_devices(device_id, "0",
+                                              threshold, vm, test)
+        virtio_mem_utils.check_numa_plugged_mem(node_id, "0",
+                                                threshold, vm, test)
+        for requested_size in requested_size_vmem_test.split():
+            req_size_normalized = int(float(normalize_data_size(requested_size,
+                                                                'B')))
+            vm.monitor.qom_set(device_id, "requested-size",
+                               req_size_normalized)
+            time.sleep(30)
+            virtio_mem_utils.check_memory_devices(device_id, requested_size,
+                                                  threshold, vm, test)
+            virtio_mem_utils.check_numa_plugged_mem(node_id, requested_size,
+                                                    threshold, vm, test)
+
+    session.close()
+    error_context.context("Shutdown guest...", test.log.debug)
+    vm.destroy()
```

Comment:
We don't need to add the following code, otherwise, the VM will boot and shut down again.

```
    error_context.context("Shutdown guest...", test.log.debug)
    vm.destroy()
```
The vt framework will help us destroy, please refer to the following code:
https://github.com/avocado-framework/avocado-vt/blob/master/virttest/env_process.py#L628-L632

Other parts ok for me:
(1/2) Host_RHEL.m9.u2.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.9.2.0.aarch64.io-github-autotest-qemu.hotplug_virtio_mem.arm64-pci: PASS (208.36 s)                                                                                                                                                                                                   (2/2) Host_RHEL.m9.u2.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.2.0.aarch64.io-github-autotest-qemu.hotplug_virtio_mem.arm64-pci: PASS (209.94 s)

Reply:
@zhenyzha thanks for the comment, so, need I to include then `kill_vm` param in the cfg file?

Reply:
Ok I see it is defined by default :grinning:

Reply:
Changed now

Reply:
@zhenyzha  @mcasquer  Destroy vm and clean env in test case is the recommend way. Framework just ensure this in post step again.   VM will not be boot again to destory when add `vm.destroy` in test case.    @zhenyzha  could you explain the reason for this comment ?

Reply:
@yanan-fu I upload detailed log to bug for explanation.

Reply:
`kernel_extra_params_add`/`kernel_extra_params_remove` will be reverted finally so VT will launch the VM again.
https://github.com/avocado-framework/avocado-vt/blob/06b7aac4be0c1080c9317abfb1468bbeb88a520d/virttest/env_process.py#L597-L607

Reply:
Ah, I see. In this way, i agree with keep vm running to restore the kernel params and framework help destroy vm, thanks!